commcare_export/writers.py

```diff
@@ -375,7 +375,13 @@ def best_type_for(self, val):
                 else:
                     return sqlalchemy.UnicodeText(collation=self.collation)
             elif self.is_mssql:
-                return sqlalchemy.NVARCHAR(collation=self.collation)
+                # MSSQL (pre 2016) doesn't allow indices on columns longer than 900 bytes
+                # - https://docs.microsoft.com/en-us/sql/t-sql/statements/create-index-transact-sql
+                # If any of our data is bigger than this, then set the column to NVARCHAR(max)
+                # `length` here is the size in bytes - https://docs.sqlalchemy.org/en/13/core/type_basics.html#sqlalchemy.types.String.params.length
+                length_in_bytes = len(val.encode('utf-8'))
+                column_length_in_bytes = None if length_in_bytes > 900 else 900
+                return sqlalchemy.NVARCHAR(length=column_length_in_bytes, collation=self.collation)
```

Comment:
I believe this will resize columns of everyone's DB currently in the wild. Maybe I should check somewhere that we aren't _shrinking_ columns? Is that even problematic?

Reply:
Resizing up seems OK, would be good to add a test to make sure it doesn't resize down.

Reply:
853fe0f0dea6b21af5c37b8a2cc729b895d48067